Adds new clown mobs for syndicate and admin use

Pugnacious-Gnome opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
This PR adds several clown subtypes, an alternate version of the honkrender that spawns clown hulks (a subtype added in this PR), a clown spawning beacon/nest, and two types of clown spawning grenades. These new entities aren't available to players by default, but admins can spawn them in if they desire to add an extra element of pranking. The dmi for the clown mobs contains a fair number of additional clown sprites not used for any current entity. I decided to include them for the convenience of other coders.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
While the jovial zeitgeist of the clown is expressed in many areas of the game, this new code allows admins to give it the spotlight, surprising players both new and old with a fresh and horrifying take on one of the game's most loved characters. This PR is also perfect for admin April fools events. 
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: several new clown types of varying strength 
add: spawner beacon/nest to spawn clown mobs
add: alternate version of the honkrender that spawns 5 clown hulks
add: new spawner grenade called C.L.U.W.N.E. which spawns a random clown and can be purchased for 5 tc as an uplink item
add: new spawner grenade called stuffed C.L.U.W.N.E. which spawns 5 mutant clowns
imageadd: sprites for new clown mobs, grenades, and spawner as well as some additional clown sprites for future mobs/admemery
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Pugnacious-Gnome committed:
new clown mobs and spawners

adds several new types of clown mobs, a beacon to spawn them, and a honkrender that spawns clown hulks

Pugnacious-Gnome committed:
added more clown varieties and fixes to previous

tralezab commented on code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:
```diff
@@ -44,3 +44,222 @@
+	name = "Fleshclown"
+	desc = "A being forged out of the pure essence of pranking, cursed into existence by a cruel maker."
+	icon = 'icons/mob/Clown_mobs.dmi'
+	icon_state = "no no no no no no no no no no no no no"
```
no, don't do this

ExcessiveUseOfCobblestone replied:
goof alt detected

tralezab commented on code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:
```diff
@@ -44,3 +44,222 @@
+	del_on_death = 1
+	loot = list(/obj/item/clothing/suit/hooded/bloated_human, /obj/item/clothing/mask/gas/clown_hat, /obj/effect/gibspawner/human, /obj/item/soap)
+
+/mob/living/simple_animal/hostile/retaliate/clown/ClownHulk
```
you should make "ClownHulk" lowercase

tralezab commented on code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:
```diff
@@ -44,3 +44,222 @@
+	mob_biotypes = list(MOB_ORGANIC, MOB_HUMANOID)
+	turns_per_move = 5
+	response_help = "tries awkwardly to hug"
+	response_disarm = "pushes the unweildy frame of"
```
```suggestion
	response_disarm = "pushes the unwieldy frame of"
```

tralezab commented:
the other thing i'd say is that these mobs aren't that unique mechanically, but whatever. i like the theme and the spawner portal laughing larry is sick

Ordonis commented:
H O N K 
O N K H
N K H O
K H O N

Very good PR please speedmerge and flood station with clown abominations.

Pugnacious-Gnome commented:
This is my first PR and my first experience coding in ss13. I'd like to work to improve it. As tralezab mentioned, there's not much variety in the mechanics of the mobs. I'm open to suggestions, though what I've submitted is about the extent of what I'm able to code myself.

KomradeSpectre commented:
For your first PR (a few code standard related things aside), it's pretty good. Your sprites are both funny and just the right amount of detail. They're also varied and imaginative. All in all, good PR!

Pugnacious-Gnome commented:
Thanks for the feedback! I agree that the mobs aren't very unique. They're mostly just reskins with some values flipped. I'm brand new to coding and still figuring out how things work, and I regret not being able to differentiate much mechanically between the mobs. I don't know if they all have enough of a justification for existing right now. Do you have any ideas for ways I could alter the code to improve this? I think I may have to look for coauthors to help, but if there's anything specific you think would help I'd love to know.

Sent from my iPhone

> On Mar 25, 2019, at 11:03 PM, tralezab <notifications@github.com> wrote:
> 
> the other thing i'd say is that these mobs aren't that unique mechanically, but whatever. i like the theme and the spawner portal laughing larry is sick
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

Arianya commented:
For admin reference, could you give an idea as to how murderous vs pranking each variant is?

ATH1909 commented on code/game/objects/items/grenades/spawnergrenade.dm:
```diff
@@ -45,3 +45,19 @@
+
+/obj/item/grenade/spawnergrenade/clown_broken
+	name = "stuffed C.L.U.W.N.E"
+	desc = "A sleek device often given to clowns on their 10th birthdays for protection.While a typical C.L.U.W.N.E only holds one creature, sometimes foolish young clowns try to cram more in, ofte to disasterous effect."
```
You forgot a space after this period.

ATH1909 commented on code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:
```diff
@@ -44,3 +44,222 @@
+	turns_per_move = 5
+	response_help = "skeptically pokes"
+	response_disarm = "pushes the unweildy frame of"
+	response_harm = "Robusts"
```
Should "Robusts" be capitalized?

tralezab replied:
No

DaxDupont commented:
these are fucking horrifying i love it

Pugnacious-Gnome commented:
The fleshclown is the least dangerous, followed by the long face. The others are a similar level of deadly, which isn't terrible, but which could really do some damage if you spawn in a bunch. The only one that's a great deal stronger than the rest is the destroyer, who smashes rwalls and does like 40 or 50 damage 

Sent from my iPhone

> On Mar 26, 2019, at 4:46 AM, Arianya <notifications@github.com> wrote:
> 
> For admin reference, could you give an idea as to how murderous vs pranking each variant is?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

tralezab commented on code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:
```diff
@@ -44,3 +44,222 @@
+	speak_chance = 20
+	a_intent = INTENT_HARM
+	maxHealth = 130
+	health = 200
```
making health higher than maxhealth is probably a bad idea

Pugnacious-Gnome replied:
whoops, thanks for pointing that out

HomocidlePancakes commented:
What does C.L.U.W.N.E stand for?

tralezab commented:
And why doesn't it have a period at the end

Pugnacious-Gnome commented:
> What does C.L.U.W.N.E stand for?

Containment and Loaded Unpacking of Neutralized Entity. It's a clown creature that's been sedated and broken down into it's base particles and atomic code. Matter is stored in a bluespace chamber and reconfigured during the unpacking phase. the stuffed C.L.U.W.N.E. has had more than one clown put into it, bugging out the unpacking system and releasing several clown mutants instead of one preserved clown creature.

mattroks101 commented on code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:
```diff
@@ -44,3 +44,222 @@
+/mob/living/simple_animal/hostile/retaliate/clown/fleshclown
+	name = "Fleshclown"
+	desc = "A being forged out of the pure essence of pranking, cursed into existence by a cruel maker."
+	icon = 'icons/mob/Clown_mobs.dmi'
```
`Clown_mobs` should probably be lowercase.

mattroks101 commented on code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:
```diff
@@ -44,3 +44,222 @@
+	harm_intent_damage = 50
+	melee_damage_lower = 40
+	melee_damage_upper = 60
+	attacktext = "Acts out divine vengeance on"
```
"Acts" should be lowercase.

mattroks101 commented on code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:
```diff
@@ -44,3 +44,222 @@
+	attack_sound = 'sound/items/bikehorn.ogg'
+	loot = list(/obj/item/clothing/mask/gas/clown_hat, /obj/effect/gibspawner/human, /obj/effect/particle_effect/foam, /obj/item/soap)
+
+/mob/living/simple_animal/hostile/retaliate/clown/Chlown
```
`Chlown` should be lowercase.

Pugnacious-Gnome replied:
Thanks! taken care of now

IndieanaJones commented:
Question, does this add that scary clown sprite from the one reddit post?  I was considering making a PR for it.

PKPenguin321 commented:
this pr is cursed

IndieanaJones commented:
Any clown-based PRs are pretty epic, so this passes with flying colors.  We need a clown shootem' up game in VR or clown planet to put these guys in, they're cool.

plapatin commented:
speedmerge asap

Anonmare commented on code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:
```diff
@@ -44,3 +44,222 @@
+	mob_biotypes = list(MOB_ORGANIC, MOB_HUMANOID)
+	turns_per_move = 5
+	response_help = "skeptically pokes"
+	response_disarm = "pushes the unweildy frame of"
```
*unwieldy

ishitbyabullet commented:
First up, fix your DMI. 32x32 mobs should not be with 64x64 mobs. Separate them into the proper DMI files which already exist.

Second, Chlown is literally just the chad meme and it looks dumb. If I'm going to see a mob that only has one dir sprite then it better be for good reason (morph, megafauna, etc) or because an admin did it. It just looks out of place.

ishitbyabullet commented:
1. Please fix your DMI folder. 32x32 mobs should be in a 32x32 DMI. 64x64 mobs should be in a 64x64 DMI. Pre-existing DMIs already exist for this or you can make two new ones with the same clown_mobs name but include 32 or 64.

2. Chlown just looks dumb. It's just the chad meme and only has one dir.

Militaires commented:
![image](https://user-images.githubusercontent.com/16103879/55109963-a359a100-50df-11e9-88c4-6148d09d49d7.png)
kinda quirky doe....

PKPenguin321 commented:
chlown is based, could use a movement frame where it does a beetle-like stride

Pugnacious-Gnome committed:
fixed unneeded capitalization and changed mob values

removed unneeded capitalization and text errors and altered mob attack values to create more variation. Honkmunculus now does very little damamge but injects a small amount of meth.

Pugnacious-Gnome commented:
> 1. Please fix your DMI folder. 32x32 mobs should be in a 32x32 DMI. 64x64 mobs should be in a 64x64 DMI. Pre-existing DMIs already exist for this or you can make two new ones with the same clown_mobs name but include 32 or 64.

I thought since the clowns are a bit of a novelty I should keep them together, sort of like the clockwork mobs or others of the like. Putting all of the graphics in the same DMI isn't an issue, as a simple offset puts them in the right square

Pugnacious-Gnome commented:
> 2. Chlown just looks dumb. It's just the chad meme and only has one dir.

clowns are already memey, and since this pr doesn't make these accessible to players without admeme trickery, I thought something like chlown could be fun. The sprite wouldn't translate well from other sides, as the chad meme is really only shown in one pose

Pugnacious-Gnome committed:
Update clown.dm

ishitbyabullet commented:
Yeah, I know it is only shown in one pose so I'm not going to even remotely suggest that you just try to somehow reinvent the meme in 3 different directions since that just sounds like hell, but I still think the sprite looks dumb personally.

ninjanomnom commented on code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:
```diff
@@ -44,3 +44,242 @@
+/mob/living/simple_animal/hostile/retaliate/clown/fleshclown
+	name = "Fleshclown"
+	desc = "A being forged out of the pure essence of pranking, cursed into existence by a cruel maker."
+	icon = 'icons/mob/clown_mobs.dmi'
```
You're repeating a lot of variables for each of these types, use a shared parent to reduce that a bit

Pugnacious-Gnome replied:
Alright. I'll reduce that in the next commit, as well as banana improvements

Pugnacious-Gnome replied:
I was already able to cut it down a little just by reverting a couple values to match the original clown, though with a little effort I can make an additional parent

Pugnacious-Gnome replied:
I'm going to make some of them have clownhulk as a parent and make some of the others closer to regular clown

Pugnacious-Gnome replied:
I fixed a lot of the redundant variables. I can do more if that's not enough

ishitbyabullet commented:
> Putting all of the graphics in the same DMI isn't an issue, as a simple offset puts them in the right square
That does sound like it would work however I think it would just be easier for all of us if it was in two different folders, to be honest.

Pugnacious-Gnome commented:
> chlown is based, could use a movement frame where it does a beetle-like stride

giving chlown beetle like stride in next commit

Pugnacious-Gnome committed:
clown changes and new clown

new clown that spawns bananas. can leave large stacks if left unattended, so it may require some reworking. nerfed several overpowered clowns and gave chlown a beetle like stride

Pugnacious-Gnome committed:
added cluwne to uplink, fixed redundant variables, added new clowns

cluwne can now be purchased for 5 telecrystals as a clown exclusive uplink item. Bananas were greatly improved. new weak clown added. Gave some clowns extra parents to reduce variables.

ExcessiveUseOfCobblestone commented on code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:
```diff
@@ -44,3 +44,263 @@
+		new /obj/item/grown/bananapeel/(pick(adjacent))
+		banana_time = world.time + rand(30,60)
+
+/mob/living/simple_animal/hostile/retaliate/clown/honkling
```
subtype this to banana and put the funjuice var at the bottom

ExcessiveUseOfCobblestone commented on code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:
```diff
@@ -44,3 +44,263 @@
+	if(banana_time < world.time)
+		var/turf/T = get_turf(src)
+		var/list/adjacent =  T.GetAtmosAdjacentTurfs(1)
+		new /obj/item/grown/bananapeel/bluespace(pick(adjacent))
```
since this is a different type of banana, we can just put a var on the mob :)

ExcessiveUseOfCobblestone commented on code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:
```diff
@@ -44,3 +44,263 @@
+		banana_time = world.time + rand(30,60)
+
+/mob/living/simple_animal/hostile/retaliate/clown/honkling
+	var/datum/reagent/funjuice = "laughter"
```
this is not a reagent, it's a string.

ExcessiveUseOfCobblestone replied:
If you're not making it dynamic I don't see why bother even having this var though, just put "laughter" in the add_reagents proc.

ExcessiveUseOfCobblestone commented on code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:
```diff
@@ -44,3 +44,263 @@
+/mob/living/simple_animal/hostile/retaliate/clown/lube
+	name = "Living Lube"
+	desc = "A puddle of lube brought to life by the honkmother."
+	icon = 'icons/mob/clown_mobs.dmi'
```
move the original clown sprite to this file, set the original clown mob's `icon` to this file, then remove this line from every subtype

ExcessiveUseOfCobblestone replied:
(not required but will save lines)

ExcessiveUseOfCobblestone commented on code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:
```diff
@@ -44,3 +44,263 @@
+
+/mob/living/simple_animal/hostile/retaliate/clown/clownhulk/chlown
+	name = "Chlown"
+	desc = "A real lunkhead who somehow gets all the girls"
```
```suggestion
	desc = "A real lunkhead who somehow gets all the girls."
```

ExcessiveUseOfCobblestone committed:
Update code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm

Co-Authored-By: Pugnacious-Gnome <48869928+Pugnacious-Gnome@users.noreply.github.com>

optimumtact commented on code/modules/uplink/uplink_items.dm:
```diff
@@ -1717,6 +1717,14 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 	item = /obj/item/storage/box/hug/reverse_revolver
 	restricted_roles = list("Clown")
 
+/datum/uplink_item/role_restricted/cluwne
```
I think this PR is fine, but I don't want these as a syndicate item.

Given the mutated/weird nature of these, I think they should remain for admins to use in portal storms and special events.

I would also allow a clown science station space ruin with a few of the grenades etc

but it's my view that this makes no sense as a syndicate item

Cyprex commented:
You could add spawner items for these for clown ops, see how the existing restrictions work in uplink_items.dm

ATH1909 commented on code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:
```diff
@@ -44,3 +44,263 @@
+		var/turf/T = get_turf(src)
+		var/list/adjacent =  T.GetAtmosAdjacentTurfs(1)
+		new /obj/item/grown/bananapeel/(pick(adjacent))
+		banana_time = world.time + rand(30,60)
```
might want to make this not spawn a banana peel on a tile that already has a banana peel to help prevent the lag that a creation passively dropping an item on the ground every 3-6 seconds would cause

ATH1909 commented on code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:
```diff
@@ -44,3 +44,263 @@
+	. = ..()
+	if(banana_time < world.time)
+		var/turf/T = get_turf(src)
+		var/list/adjacent =  T.GetAtmosAdjacentTurfs(1)
```
here too

ATH1909 commented on code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:
```diff
@@ -44,3 +44,263 @@
+	turns_per_move = 1
+	speed = -10
+	harm_intent_damage = 1
+	melee_damage_lower = 1
```
could you reduce both this and melee_damage_upper to 0, or does the code/game break if that happens?

ExcessiveUseOfCobblestone replied:
it uses friendly verbs if you're 0

Pugnacious-Gnome replied:
I also have a proc set to check when it attacks, and it doesn't activate when it's set to 0

ATH1909 commented on code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm:
```diff
@@ -44,3 +44,263 @@
+	speed = -5
+	harm_intent_damage = 5
+	melee_damage_lower = 5
+	melee_damage_upper = 10
```
Maybe this kind of clown should also deal no direct damage on a strike, making it more of a "support clown"? It could still hurt/kill people by ODing them on and getting them addicted to meth, of course.

ATH1909 commented:
Idea here: If a clown (or one of the various clown simplemobs) gets hit by a wabbajack bolt, they get turned into one of these clown simplemobs (or the normal clown simplemob) instead of being turned into something from the normal wabbajack mob pool.

ExcessiveUseOfCobblestone commented:
10 days post a maintainer review with no feedback

PKPenguin321 commented:
please fix this and reopen it i need it

DaxDupont commented:
if this PR isn't reopened by friday i'll steal it and fix it up
